classes/local/datasource/base.php

```diff
@@ -245,6 +245,13 @@ public function get_cached_data(): \stdClass {
         return $data;
     }
 
+    /**
+     * Get the cache used by the datasource.
```

Comment:
Would be useful to add some a bit more documentation here to help developers to understand what sort of cache should bed returned by an implementation and how it will be used by mod_cms.